SA3/qa_web/views.py

```diff
@@ -102,6 +102,48 @@ def answers(request, id_):
     return render(request, 'qa_web/answerspage.html', {'currentQuestion': q, 'answers': q_answers, 'bestAnswer': q_best_answer})
 
 
+def vote(request):
+    """Receives Ajax queries to vote on Posts, updates the corresponding Post and returns the new score"""
+    if request.method == 'POST' and request.user.is_authenticated:
+        vote_direction, button_id, post_type = request.POST['button'].split('_')
```

Comment:
Yes! Yes! Yes! Such a smart way to obtain all the data from `POST` request! I actually love this line, gotta love Python.